core/unit_test/TestBitManipulation.cpp

```diff
@@ -429,4 +429,53 @@ constexpr auto test_bit_width(UInt x) -> decltype(Kokkos::bit_width(x)) {
 TEST_BIT_MANIPULATION(bit_width);
 //</editor-fold>
 
+//<editor-fold desc="[bit.byteswap]">
+template <class T>
+constexpr auto test_byteswap(T x) -> decltype(Kokkos::byteswap(x)) {
+  using Kokkos::byteswap;
+
+  static_assert(noexcept(byteswap(x)));
+  static_assert(std::is_same_v<decltype(byteswap(x)), T>);
+
+  return true;
+}
+
+constexpr X test_byteswap(...) { return {}; }
+
+static_assert(test_byteswap((void*)0).did_not_match());  // NOLINT
+static_assert(test_byteswap((float)0).did_not_match());
+constexpr char c2[2] = {};
+static_assert(test_byteswap(c2).did_not_match());
+static_assert(test_byteswap((char)0));
+static_assert(test_byteswap((short)0));
+static_assert(test_byteswap((int)0));
+static_assert(test_byteswap((long)0));
+static_assert(test_byteswap((long long)0));
+static_assert(test_byteswap((unsigned char)0));
+static_assert(test_byteswap((unsigned short)0));
+static_assert(test_byteswap((unsigned int)0));
+static_assert(test_byteswap((unsigned long)0));
+static_assert(test_byteswap((unsigned long long)0));
+
+constexpr bool test_byteswap2() {
+  using Kokkos::byteswap;
+
+  static_assert(byteswap<int8_t>(0x12) == 0x12);
+  static_assert(byteswap<int16_t>(0x1234) == 0x3412);
+  static_assert(byteswap<int32_t>(0x12345678) == 0x78563412);
+
+  static_assert(byteswap<int64_t>(0x123456789abcdef0ll) ==
+                0xf0debc9a78563412ll);
```

Comment:
Having learned about the standardized macros `INT64_C` etc al, I'm inclined to favor their use over the suffixes, since the macros specify a size rather than the annoying minimums of the primitive types.

Reply:
@PhilMiller I'll make that change.  (While I knew there had to be some mechanism for it, I just learned about the macros two days ago.)